src/base/async/int.c

```diff
@@ -1929,100 +1927,25 @@ static void redirect_devices(void)
 
 static int redir_it(void)
 {
-    uint16_t lol_lo, lol_hi, sda_lo, sda_hi, sda_size, redver, mosver;
-    uint8_t major, minor;
-    int is_MOS;
-    int is_cf;
-
     if (redir_state)
 	return 0;
     redir_state++;
 
-    /*
-     * To start up the redirector we need
-     * (1) the list of list,
-     * (2) the DOS version and
-     * (3) the swappable data area.
-     */
     pre_msdos();
 
-    LWORD(eax) = 0x1100;
-    do_int_call_back(0x2f);
-    if (LO(ax) != 0xff) {
-	error("Redirector unavailable\n");
-	post_msdos();
-	return 0;
-    }
-
-    LWORD(eax) = 0x5200;
-    call_msdos();
-    ds_printf
-	("INT21 +1 at %04x:%04x: AX=%04x, BX=%04x, CX=%04x, DX=%04x, DS=%04x, ES=%04x\n",
-	 SREG(cs), LWORD(eip), LWORD(eax), LWORD(ebx),
-	 LWORD(ecx), LWORD(edx), SREG(ds), SREG(es));
-    lol_lo = LWORD(ebx);
-    lol_hi = SREG(es);
-
-    LWORD(eax) = LWORD(ebx) = LWORD(ecx) = LWORD(edx) = 0x3099; // MOS version trigger
-    call_msdos();
-    ds_printf
-	("INT21 +2b at %04x:%04x: AX=%04x, BX=%04x, CX=%04x, DX=%04x, DS=%04x, ES=%04x\n",
-	 SREG(cs), LWORD(eip), LWORD(eax), LWORD(ebx),
-	 LWORD(ecx), LWORD(edx), SREG(ds), SREG(es));
-    mosver = LWORD(eax);
-    LWORD(eax) = 0x3000;                                        // DOS version std request
-    call_msdos();
-    ds_printf
-	("INT21 +2 at %04x:%04x: AX=%04x, BX=%04x, CX=%04x, DX=%04x, DS=%04x, ES=%04x\n",
-	  SREG(cs), LWORD(eip), LWORD(eax), LWORD(ebx),
-	 LWORD(ecx), LWORD(edx), SREG(ds), SREG(es));
-    major = LO(ax);
-    minor = HI(ax);
-    is_MOS = (LWORD(eax) != mosver); // different result!
+    LWORD(eax) = DOS_HELPER_MFS_HELPER;
+    LWORD(ebx) = DOS_SUBHELPER_MFS_REDIR_INIT;
+    mfs_inte6();
```

Comment:
There's only one caller of `redir_it()`, that's `dos_post_boot()` and that's protected from multiple calls by the `post_boot` variable. Do I need some more protection than that?

Reply:
Yes but `DOS_SUBHELPER_MFS_REDIR_INIT`
can still be called more than once because of
`emufs.sys`. If you are going to move `emufs_loaded`
to int.c, then you should also use it here as a check.

Reply:
I think this was the reason why `emufs_loaded`
was initially in mfs.c: likely because the init was
supposed to be called from 2 unrelated places,
and so the synchronization on a callee site makes
sense. But since `redir_it()` is a hack that should
go away, I will not object against moving the
protection to int.c.